padertorch/modules/normalization.py

```diff
@@ -1,104 +1,337 @@
-                axis=self.statistics_axis,
-                keepdims=True
+        self.reset_running_stats()
+        if self.gamma is not None:
+            nn.init.ones_(self.gamma.scale)
+        if self.beta is not None:
+            nn.init.zeros_(self.beta.shift)
+
+    def forward(self, x, sequence_lengths=None):
+        if self.training or not self.track_running_stats:
+            y, mean, power, n_values = normalize(
+                x, gamma=self.gamma, beta=self.beta,
+                statistics_axis=self.statistics_axis,
+                batch_axis=self.batch_axis, sequence_axis=self.sequence_axis,
+                sequence_lengths=sequence_lengths, shift=self.shift, scale=self.scale,
+                eps=self.eps
             )
+            if self.track_running_stats:
+                self.num_tracked_values += n_values.data
+                if self.momentum is None:
+                    momentum = 1 - n_values / self.num_tracked_values.data
+                else:
+                    momentum = self.momentum
+                if self.shift:
+                    self.running_mean *= momentum
+                    self.running_mean += (1 - momentum) * mean.data
+                    power = power.data + mean.data ** 2
+                if self.scale:
+                    self.running_power *= momentum
+                    self.running_power += (1 - momentum) * power.data
+                if self.interpolation_factor > 0.:
+                    # perform straight through backpropagation
+                    # https://arxiv.org/pdf/1611.01144.pdf
+                    y_ = x
+                    if self.shift:
+                        y_ = y_ - self.running_mean.data
+                    if self.scale:
+                        y_ = y_ / torch.sqrt(self.runnning_var.data)
+                    y = y + self.interpolation_factor * (y_ - y).detach()
+                    y = y * compute_mask(y, sequence_lengths, self.batch_axis, self.sequence_axis)
         else:
-            raise ValueError(f'chosen order {self.order} in is not'
-                             f' known in {self}')
+            y = x
+            if self.shift:
+                y = y - self.running_mean.data
+            if self.scale:
+                y = y / torch.sqrt(self.runnning_var.data)
+            if self.gamma is not None:
+                y = y * self.gamma
+            if self.beta is not None:
+                y = y + self.beta
+            y = y * compute_mask(x, sequence_lengths, self.batch_axis, self.sequence_axis)
+        return y
+
+    def inverse(self, x):
+        if not self.track_running_stats:
+            raise NotImplementedError
+        if self.beta is not None:
+            x = x - self.beta
+        if self.gamma is not None:
+            x = x / self.gamma
+        if self.scale:
+            x = torch.sqrt(self.running_var) * x
+        if self.shift:
+            x = x + self.running_mean
+        return x
+
+
+class Normalize(Function):
```

Comment:
perhaps you should change this to a private class, since the user is supposed to use the Norm class.
The same holds for the normalize function

Reply:
another option would be to change the name to clarifiy that this is just an intermediate class

Reply:
If I understood this correctly, the `normalize` function can still be used on its own if you don't want it to show up in the Model's repr, don't need any statistics tracking, and don't like the string axis definition.

Reply:
yes, that is true but as a default people should be encouraged to use the Norm class and only use the normalize function if they know what they are doing

Reply:
and even if you are using the normalize function the Normalize class is still hidden